backend/models/accountPermissions.js

```diff
@@ -0,0 +1,102 @@
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+"use strict";
+const responseCodes = require("../response_codes.js");
+const C = require("../constants");
+const { intersection } = require("lodash");
+const Project = require("./project");
+
+const updatePermissions = async function(teamspace, updatedPermissions) {
+	const User = require("./user");
+	await User.update(teamspace.user, {"customData.permissions": updatedPermissions});
+	return updatedPermissions;
+};
+
+const checkValidUpdate = async (teamspace, teamMember, permissions) => {
+	const User = require("./user");
+	const userToCheck = await User.findByUserName(teamMember);
+	if(!userToCheck) {
+		throw (responseCodes.USER_NOT_FOUND);
+	}
+
+	if(!User.isMemberOfTeamspace(userToCheck, teamspace.user)) {
+		throw(responseCodes.USER_NOT_ASSIGNED_WITH_LICENSE);
+	}
+
+	const isPermissionInvalid = permissions &&
+		intersection(permissions, C.ACCOUNT_PERM_LIST).length !== permissions.length;
+
+	if (isPermissionInvalid) {
+		throw(responseCodes.INVALID_PERM);
+	}
+};
+
+const AccountPermissions = {};
+
+AccountPermissions.findByUser = function(user, username) {
+	return this.get(user).find(perm => perm.user === username);
+};
+
+AccountPermissions.get = function(teamspace) {
+	return  (teamspace.customData || {}).permissions || [];
+};
+
+AccountPermissions.updateOrCreate = async function(teamspace, username, permissions) {
+	await checkValidUpdate(teamspace, username, permissions);
+
+	if(permissions && permissions.length === 0) {
+		// this is actually a remove
+		return await this.remove(teamspace, username);
+	}
+
+	const updatedPermissions = this.get(teamspace).filter(perm => perm.user !== username)
+		.concat({user: username, permissions});
+
+	return await updatePermissions(teamspace, updatedPermissions);
+};
+
+AccountPermissions.update = async function(teamspace, username, permissions) {
+	await checkValidUpdate(teamspace, username, permissions);
+
+	const currPermission = this.findByUser(teamspace, username);
+
+	if(!currPermission) {
+		throw (responseCodes.ACCOUNT_PERM_NOT_FOUND);
+	}
+
+	return await this.updateOrCreate(teamspace, username, permissions);
+};
+
+AccountPermissions.remove = async function(teamspace, userToRemove) {
+	const updatedPermissions = this.get(teamspace).filter(perm => perm.user !== userToRemove);
+
+	if (updatedPermissions.length >=  this.get(teamspace).length) {
+		throw responseCodes.ACCOUNT_PERM_NOT_FOUND;
+	}
+
+	const projects = await Project.find({ account: teamspace.user },{ "permissions.user": userToRemove});
```

Comment:
There should probably be a method in Project to remove a user from the project, instead of account Permissions depicting everything

tagging @Charence  since he's doing project so he knows on his branch!